### PR TITLE
kernel: Config changes to converge kernel-hci with kernel

### PR DESCRIPTION
**Pull request by vinceaperri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR is the first of a two-phase plan to remove the kernel-hci package and direct consumers of this package to using the kernel package. This phase completely aligns the kernel configs of the two packages, and also aligns the kernel configs changes for amd64 with aarch64.

###### Change Log  
- Add/update config options in the kernel config that were explicitly set/required in the kernel-hci package across multiple PRs
- Add/update config in mariner-required-configs.json accordingly, linking to the original PRs for the kernel-hci package where applicable.

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: [496340](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=496340&view=results)
- Boot kernel RPM in a VM

### CODE REVIEW

File: toolkit/scripts/mariner-required-configs.json [hunk 1]
```diff
@@ -1387,7 +1392,306 @@
...
+            "CONFIG_INET_TCP_DIAG": {
+                "value": [
+                    "m"
+                ],
+                "arch": [
+                    "AMD64"
+                ],
+                "comment": "Needed for HCI scenarios",
+                "PR": [
+                    "https://github.com/microsoft/CBL-Mariner/pull/####",
```
Comment (PawelWMS) [inline toolkit/scripts/mariner-required-configs.json:1483]: Are the `pull/####` lines correct?

**Review by PawelWMS — APPROVED** [review]
Approving the parts, that require approval from the @CBL-Mariner-Tooling group. I think otherwise it might be blocked. It looks like it requires expertise of the kernel team, though.

And one request: could you fill out the PR's summary with more details, so we have a good description for now and a trace for the future?

File: SPECS/kernel/config [hunk 2]
```diff
@@ -7278,7 +7286,7 @@ CONFIG_HARDENED_USERCOPY=y
 CONFIG_FORTIFY_SOURCE=y
 # CONFIG_STATIC_USERMODEHELPER is not set
 CONFIG_SECURITY_SELINUX=y
-# CONFIG_SECURITY_SELINUX_BOOTPARAM is not set
+CONFIG_SECURITY_SELINUX_BOOTPARAM=y
```
Comment (rlmenge) [inline SPECS/kernel/config:7289]: Does this require `SECURITY_SELINUX_BOOTPARAM_VALUE` to be set as well?

File: SPECS/kernel/config [hunk 3]
```diff
@@ -1008,7 +1008,7 @@ CONFIG_COMPAT_BINFMT_ELF=y
 CONFIG_ELFCORE=y
 CONFIG_CORE_DUMP_DEFAULT_ELF_HEADERS=y
 CONFIG_BINFMT_SCRIPT=y
-CONFIG_BINFMT_MISC=y
+CONFIG_BINFMT_MISC=m
```
Comment (rlmenge) [inline SPECS/kernel/config:1011]: Could you copy these changes to aarch64 to keep the configs aligned?

**Review by rlmenge — CHANGES_REQUESTED** [review]

File: SPECS/kernel/config (hunk 2 above)
Comment (vinceaperri) [reply SPECS/kernel/config:7289]: I do not see `SECURITY_SELINUX_BOOTPARAM_VALUE` in the code. We have only been setting `CONFIG_SECURITY_SELINUX_BOOTPARAM` to enable the kernel command line parameter. It allows images to be built with selinux built-in but disabled or in permissive mode and does not change functionality should the parameter not be used.

File: SPECS/kernel/config (hunk 3 above)
Comment (vinceaperri) [reply SPECS/kernel/config:1011]: Done. For all the config options under consideration, I have made sure the same value is reflected in aarch64. I have also add ARM64 for all the config options too. Please let me know if that is what you intended.

File: SPECS-SIGNED/kernel-signed/kernel-signed.spec [hunk 4]
```diff
@@ -153,6 +153,12 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %exclude /module_info.ld
 
 %changelog
+* Thu Feb 01 2024 Vince Perri <viperri@microsoft.com> - 6.6.12.1-4
+- Bump release to match kernel
+
```
Comment (rlmenge) [inline SPECS-SIGNED/kernel-signed/kernel-signed.spec:158]:
```suggestion
```

Remove duplicate changelog entry

File: SPECS/kernel/config (hunk 2 above)
Comment (rlmenge) [reply SPECS/kernel/config:7289]: Ah yup, looks like `SECURITY_SELINUX_BOOTPARAM_VALUE` option was removed. Sounds good :)

File: SPECS/kernel/kernel.spec [hunk 5]
```diff
@@ -429,6 +429,12 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Thu Feb 01 2024 Vince Perri <viperri@microsoft.com> - 6.6.12.1-4
```
Comment (rlmenge) [inline SPECS/kernel/kernel.spec:432]: Duplicate

File: SPECS/kernel-headers/kernel-headers.spec [hunk 6]
```diff
@@ -75,6 +75,12 @@ done
 %endif
 
 %changelog
+* Thu Feb 01 2024 Vince Perri <viperri@microsoft.com> - 6.6.12.1-4
```
Comment (rlmenge) [inline SPECS/kernel-headers/kernel-headers.spec:78]: Duplicate

File: SPECS/kernel/kernel.signatures.json [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
- "Signatures": {
-  "cbl-mariner-ca-20211013.pem": "5ef124b0924cb1047c111a0ecff1ae11e6ad7cac8d1d9b40f98f99334121f0b0",
-  "config": "541f9228eaa516b8229138ce53e26096eae6316923e7e86ae43d6adc26f2743b",
-  "config_aarch64": "596e47471b83585839130f31ff66507a89c2827d87c763840d893a22da830a5b",
-  "cpupower": "d7518767bf2b1110d146a49c7d42e76b803f45eb8bd14d931aa6d0d346fae985",
-  "cpupower.service": "b057fe9e5d0e8c36f485818286b80e3eba8ff66ff44797940e99b1fd5361bb98",
-  "sha512hmac-openssl.sh": "02ab91329c4be09ee66d759e4d23ac875037c3b56e5a598e32fd1206da06a27f",
-  "kernel-6.6.12.1.tar.gz": "7312d1bc82f7c42c1b327258d63aeb9e473f97e42c598dd6e262baf0d80ec081"
- }
+  "Signatures": {
```
Comment (rlmenge) [inline SPECS/kernel/kernel.signatures.json:2]: Are these spaces intended?

**Review by rlmenge — CHANGES_REQUESTED** [review]

**Review by rlmenge — APPROVED** [review]

**Pull request merged**